code/app/src/main/java/com/adobe/marketing/mobile/messagingsample/CodeBasedExperienceActivity.kt

```diff
@@ -25,24 +25,35 @@ class CodeBasedExperienceActivity : AppCompatActivity() {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_codebased)
 
-        // retrieve any cached code based experiences
         var propositions = mutableListOf<Proposition>()
-        val surfaces = mutableListOf<Surface>()
-        val surface = Surface("<your-surface-path>")
-        surfaces.add(surface)
-        Messaging.getPropositionsForSurfaces(surfaces) {
-            println("getPropositionsForSurfaces callback contained ${it.entries.size} entry/entries for surface ${surface.uri}")
+        val surfaceList = mutableListOf<Surface>()
+        surfaceList.add(Surface("<your-surface-path>"))
+
+        // fetch message feed and code based experiences
+        Messaging.updatePropositionsForSurfaces(surfaceList)
+        Thread.sleep(500)
+        // retrieve any cached code based experiences
+        Messaging.getPropositionsForSurfaces(surfaceList) {
+            println(
+                "getPropositionsForSurfaces callback contained ${it.entries.size} entry/entries for surfaces ${
+                    surfaceList.get(
+                        0
+                    ).uri
+                }"
+            )
             for (entry in it.entries) {
                 propositions = entry.value
             }
 
             // show code based experiences
             val codeBasedExperienceWebView: WebView = findViewById(R.id.codeBasedExperienceWebView)
             val htmlContentString = propositions[0].items[0].content
```

Comment:
Storing JSON content in `htmlContentString` and then trying to load it in a webview doesn't seem like something we want to prescribe. How about we refactor this a bit to show JSON content in a text view?